### DISCUSSION TITLE
Do not add trailing semicolon on a one-line JSX tag

### DISCUSSION THREAD

**Discussion by paleo** [biomejs/biome#617] [General]
Hi, I currently need to add a comment to disable Biome formatting each time I have a one-line JSX tag, like this:

```ts
  // biome-ignore format: no trailing semi-colon
  const el = <div>abc</div>
```

If I remove the comment, then Biome appends a semicolon:

```ts
  const el = <div>abc</div>;
```

… but then the semicolon is part of the template.

**Comment by unvalley** [comment]
@paleo 
If a configuration of `javascript.formatter.semicolons` is `always`, current biome formatter behaves as you said.
You can use `asNeeded` option to this to prevent the semicolon insertion, but this way would affect other codes, though.
The current formatting on `always` follows prettier formatting, so it's difficult to change quickly.

ref: https://biomejs.dev/reference/configuration/#javascriptformattersemicolons

**Comment by arthurfiorette** [comment]
Hey @paleo, the semicolon at `const el = <div>abc</div>;` **should NOT be a part of the template**, even if its present. **This is likely a bug in your JSX renderer**... JSX only allows one root child which is already occupied by your `<div>` call.

Look at this example: 

```tsx
import React from "react";

// Semicolon at the end (biome may add it)
const a = <div>a</div>;

// Example of semicolon inserted manually inside a jsx tag, (biome does not add semicolons here)
const b = (
  <>
    <div>b</div>;
  </>
);
```

gets transpiled by `tsc` to

```tsx
import React from "react";

const a = React.createElement("div", null, "a");

const b = React.createElement(
  React.Fragment,
  null,
  React.createElement("div", null, "b"),
  ";", // you can see the semicolon gets registered into the template call.
);
```

Only if you have your semicolon inside the template it will be added into the final output.

**Reply by paleo** [reply]
Thanks! And now I let Biome adding the semicolons and it works fine. I had the problem before, though. But maybe it was a bug. I close the ticket.
